app/V1Module/router/RouterFactory.php

```diff
@@ -108,6 +108,8 @@ private static function createExercisesRoutes(string $prefix): RouteList {
     $router[] = new PostRoute("$prefix/<id>/fork", "Exercises:forkFrom");
     $router[] = new GetRoute("$prefix/<id>/pipelines", "Exercises:getPipelines");
     $router[] = new PostRoute("$prefix/<id>/hardware-groups", "Exercises:hardwareGroups");
+    $router[] = new PostRoute("$prefix/<id>/attach-group", "Exercises:attachGroup");
+    $router[] = new DeleteRoute("$prefix/<id>/detach-group", "Exercises:detachGroup");
```

Comment:
This could be a little more REST-ish - I would imagine something like PUT/DELETE `exercises/<exerciseId>/groups/<groupId>`. Another (perhaps more important) advantage would be that all the important information would be in the URL and not hidden in the request body (which is BTW illegal in DELETE requests), which is much better for checking permissions.

Reply:
Good point, will change